src/main/java/network/oxalis/as4/util/OxalisAlgorithmSuiteLoader.java

```diff
@@ -70,13 +70,14 @@ public static class OxalisAlgorithmSuite extends AlgorithmSuite {
                             128, 128, 128, 256, 1024, 4096
                     )
             );
-            ALGORITHM_SUITE_TYPES.get(BASIC_128_GCM_SHA_256_MGF_SHA_256).setMGFAlgo(MGF_SHA256);
-            ALGORITHM_SUITE_TYPES.get(BASIC_128_GCM_SHA_256_MGF_SHA_256).setEncryptionDigest(SPConstants.SHA256);
+            AlgorithmSuiteType algorithmSuiteType = ALGORITHM_SUITE_TYPES.get(BASIC_128_GCM_SHA_256_MGF_SHA_256);
+            algorithmSuiteType.setMGFAlgo(MGF_SHA256);
+            algorithmSuiteType.setEncryptionDigest(SPConstants.SHA256);
+            algorithmSuiteType.setAsymmetricSignature("http://www.w3.org/2001/04/xmldsig-more#rsa-sha256");
         }
 
         OxalisAlgorithmSuite(final SPConstants.SPVersion version, final Policy nestedPolicy) {
             super(version, nestedPolicy);
-            this.setAsymmetricSignature("http://www.w3.org/2001/04/xmldsig-more#rsa-sha256");
```

Comment:
I do not understand this change - does it relate to the topic of PR?

Reply:
The API of WSS4J have changed. The method `setAsymmetricSignature` appears to have been moved from the top level class `AlgorithmSuite` to the nested class `AlgorithmSuiteType`.

Reply:
OK, good! But could it be that AsymmetricSignature is not set to "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256" for 
BASIC_128_GCM_SHA_256
but only for 
BASIC_128_GCM_SHA_256_MGF_SHA_256?

In previous version, it was defined on a  OxalisAlgorithmSuite level, in a new version - on AlgorithmSuiteType, but in https://github.com/OxalisCommunity/Oxalis-AS4/blob/824db3139283f27fdf5eba98e24f37e56b9015bf/src/main/java/network/oxalis/as4/util/OxalisAlgorithmSuiteLoader.java#L76 it is set only for BASIC_128_GCM_SHA_256_MGF_SHA_256, right?

May be we miss similar line for BASIC_128_GCM_SHA_256?

If we make mistake here, it will be quite difficult to find it out...

Reply:
Yes. The asymmetric signature is now only set for the second custom `AlgorithmSuiteType`, BASIC_128_GCM_SHA_256_MGF_SHA_256. It might well be that it is needed for the first one, too.

I put it on the second suite type, since it already had a couple of customizations, guessing that the first one might not need customisation. But I don't know these WSS4J APIs, I just did the bare minimum to get it to compile against newer CXF and WSS4J. And it appears to be working in our test environment, but that might be a false positive, if some default asymmetric signature with weaker security is used instead...

Do you think its safer to assign the asymmetric signature on both?

Reply:
From the WSS4J code, the algorithm suite gets a default asymmetric signature, when we don't specify it. So it's probably safest to set it on both suites. I have added that in the last commit.

Reply:
Thank you! Yes, this looks safer - better to keep the same logic as much as possible during upgrades.
I think it is ok now - although I would write it like before, without extracting to a variable algorithmSuiteType for only one of suites:

```
ALGORITHM_SUITE_TYPES.get(BASIC_128_GCM_SHA_256_MGF_SHA_256).setMGFAlgo(MGF_SHA256);
ALGORITHM_SUITE_TYPES.get(BASIC_128_GCM_SHA_256_MGF_SHA_256).setEncryptionDigest(SPConstants.SHA256);
ALGORITHM_SUITE_TYPES.get(BASIC_128_GCM_SHA_256_MGF_SHA_256)
	.setAsymmetricSignature("http://www.w3.org/2001/04/xmldsig-more#rsa-sha256");

ALGORITHM_SUITE_TYPES.get(BASIC_128_GCM_SHA_256)
	.setAsymmetricSignature("http://www.w3.org/2001/04/xmldsig-more#rsa-sha256");
```
In this case, the commit diff would clearly highlight, that you just added .setAsymmetricSignature("http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"); to both suite types because it is removed from Suite.
But it is just a question of styling :)

Anyway, let's see @aaron-kumar review too :)

Reply:
Ok, seems like @aaron-kumar did a superset of these changes in the latest release, and also added code to adjust the signature algorithm.